WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -191,11 +233,29 @@ private extension CardReaderConnectionController {
                 ///
                 self.foundReaders = cardReaders
                 self.pruneSkippedReaders()
+                self.updateShowSeveralFoundReaders()
 
-                /// This completion can be called repeatedly as more readers
-                /// become discovered. To avoid interrupting connecting to
-                /// a known reader, or interrupting the user prompt for a unknown
-                /// reader, ensure we are in the searching state first
+                /// Note: This completion will be called repeatedly as the list of readers
+                /// discovered changes, so some care around state must be taken here.
+                ///
+
+                /// If the found-several-readers view is already presenting, update its list of found readers
+                ///
+                if case .foundSeveralReaders = self.state {
+                    self.alerts.updateSeveralReadersList(readerIDs: self.getFoundReaderIDs())
+                }
+
+                /// If we should switch from a single found reader prompt to several, do so now
+                ///
+                if case .foundReader = self.state {
```

Comment:
I know this is irrelevant, but I am never going to be able to parse this syntax

Reply:
I have to look it up myself